regparser/layer/external_citations.py

```diff
@@ -33,13 +33,19 @@ def reformat_citation(self, citation):
         return [c for c in citation if c not in [
             'U.S.C.', 'CFR', 'part', '.', 'Public', 'Law', '-']]
 
-    def parse(self, text, parts=None):
+    def parse(self, text, parts=None, title=None):
         """ Parse the provided text, pulling out all the citations. """
         parser = grammar.regtext_external_citation
 
         cm = defaultdict(list)
         citation_strings = {}
         for citation, start, end in parser.scanString(text):
+            # Discard citations of form XX CFR YY if XX and YY are the title
+            # and part being parsed
+            if (citation[0] == self.cfr_title and
```

Comment:
Re: ignore everything that is an internal cite- As far as I can tell, the only internal citations that could potentially be mistaken for externals are CFR references, so I think it's the same thing either way but this is likely easier/ faster. Does that seem right to you?